File: android/app/src/main/java/com/cbq/nnlg/CourseForm.java
package com.cbq.nnlg;

public class CourseForm {

    private String courseName;
    private String courseClassRoom;
    private String courseTeacher;
    private String courseWeek;

    public String getCourseName() {
        return courseName;
    }

    public void setCourseName(String courseName) {
        this.courseName = courseName;
    }

    public String getCourseClassRoom() {
        return courseClassRoom;
    }

    public void setCourseClassRoom(String courseClassRoom) {
        this.courseClassRoom = courseClassRoom;
    }

    public String getCourseTeacher() {
        return courseTeacher;
    }

    public void setCourseTeacher(String courseTeacher) {
        this.courseTeacher = courseTeacher;
    }

    public String getCourseWeek() {
        return courseWeek;
    }

    public void setCourseWeek(String courseWeek) {
        this.courseWeek = courseWeek;
    }


    public String toJson(){

        StringBuffer stringBuffer = new StringBuffer();

        stringBuffer.append("{");
        if(courseName!=null)
            stringBuffer.append("\"courseName\":"+"\""+courseName+"\"");
        if(courseClassRoom!=null)
        stringBuffer.append(",\"courseClassRoom\":"+"\""+courseClassRoom+"\"");
        if(courseTeacher!=null)
            stringBuffer.append(",\"courseTeacher\":"+"\""+courseTeacher+"\"");
        if(courseWeek!=null)
            stringBuffer.append(",\"courseWeek\":"+"\""+courseWeek+"\"");
        stringBuffer.append("}");

        return stringBuffer.toString();
    }


}

File: lib/utils/edusys/entity/CourseForm.dart
import 'dart:developer';

import 'package:flutter_svg/flutter_svg.dart';

class CourseForm{
  String? courseName;
  String? courseClassRoom;
  String? courseTeacher;
  String? courseWeek;
  List<int>? courseWeekList;
  List<int>? courseSectionList;

  String? toJson() {
    String stringBuffer= "";
    stringBuffer+="{";
    // if(courseName!=null)
    //   stringBuffer+="\"courseName\":"+"\""+courseName!+"\"";
    // if(courseClassRoom!=null)
    //   stringBuffer+=",\"courseClassRoom\":"+"\""+courseClassRoom!+"\"";
    // if(courseTeacher!=null)
    //   stringBuffer+=",\"courseTeacher\":"+"\""+courseTeacher!+"\"";
    // if(courseWeek!=null)
    //   stringBuffer+=",\"courseWeek\":"+"\""+courseWeek!+"\"";
    // stringBuffer+="}";

    if(courseName!=null)
      stringBuffer+="\"courseName\":"+"\""+courseName!+"\"";
    if(courseClassRoom!=null)
      stringBuffer+=",\"courseClassRoom\":"+"\""+courseClassRoom!+"\"";
    if(courseTeacher!=null)
      stringBuffer+=",\"courseTeacher\":"+"\""+courseTeacher!+"\"";
    if(courseWeek!=null)
      stringBuffer+=",\"courseWeek\":"+"\""+courseWeek!+"\"";

    if(courseWeekList!=null){
      stringBuffer+=",\"courseWeekList\":";
      stringBuffer+="[";
      for (int i = 0; i < courseWeekList!.length; i++) {
        stringBuffer+='${courseWeekList![i]}';
        if(i!=courseWeekList!.length-1)stringBuffer+=',';
      }
      stringBuffer+=']';
    }

    if(courseSectionList!=null){
      stringBuffer+=",\"courseSectionList\":";
      stringBuffer+="[";
      for (int i = 0; i < courseSectionList!.length; i++) {
        stringBuffer+='${courseSectionList![i]}';
        if(i!=courseSectionList!.length-1)stringBuffer+=',';
      }
      stringBuffer+="]";
    }

    stringBuffer+="}";
    // log(stringBuffer);
    return stringBuffer;
  }
}